dfu: mcuboot: Rename and move dfu_ctx_mcuboot_set_b1_file

glarsennordic opened the pull request:
dfu_target -> dfu_ctx_mcuboot_set_b1_file is now
fota_download -> fota_download_parse_dual_filepath,

and unit tests have been adjusted accordingly

IRIS3988

Signed-off-by: Georges Oates_Larsen <georges.larsen@nordicsemi.no>

glarsennordic commented on tests/subsys/net/lib/fota_download/CMakeLists.txt:
```diff
@@ -1,8 +1,9 @@
 #
-# Copyright (c) 2019 Nordic Semiconductor
+# Copyright (c) 2020 Nordic Semiconductor
```
Oops. This should be 2022

glarsennordic replied:
This needs to be 2022. Will fix after reviews

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-8345
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

glarsennordic commented on tests/subsys/net/lib/fota_download/src/test_fota_download.c:
```diff
@@ -240,76 +229,87 @@ void client_callback(const struct fota_download_evt *evt)
+ * @param expected_selection - The resource locator that fota_download_start is expected to select
+ * @param s0_active - If true, pretend that s0 is active. Otherwise, pretend that s1 is active.
+ */
+static void test_fota_download_start_generic(const char * const resource_locator,
```
Git absolutely butchers the diff on this function, ignore diff and treat like a brand new function

plskeggs replied:
Yes, it's very difficult to follow.

plskeggs commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -454,7 +454,9 @@ Bluetooth libraries and services
-|no_changes_yet_note|
+* :ref:`lib_dfu_target` library:
+
+   * Removed :c:func:`dfu_ctx_mcuboot_set_b1_file` (converted to a util function in :ref:`lib_fota_download`)
```
Instead, you could say:
* Moved :c:func:`dfu_ctx_mcuboot_set_b1_file` to :ref:lib_fota_download` and renamed to...

plskeggs commented on tests/subsys/net/lib/fota_download/src/test_fota_download.c:
```diff
@@ -11,24 +12,19 @@
 
 /* Stubs and mocks */
-bool dfu_ctx_mcuboot_set_b1_file__s0_active;
 const char *download_client_start_file;
```
Should `download_client_start_file` and `dfu_ctx_mcuboot_set_b1_file__update` need to be public?  I know your PR didn't add these, but maybe should be static anyway.

plskeggs reviewed (approved):
Seems fine to me, assuming the tests pass.

peknis commented on subsys/net/lib/fota_download/include/fota_download_util.h:
```diff
@@ -0,0 +1,38 @@
+/**
+ * @brief Find correct MCUBoot update resource locator entry in space separated string.
+ *
+ * When supporting upgrades of the MCUBoot bootloader, two variants of the new firmware is
```
```suggestion
 * When supporting upgrades of the MCUBoot bootloader, two variants of the new firmware are
```

peknis commented on subsys/net/lib/fota_download/include/fota_download_util.h:
```diff
@@ -0,0 +1,38 @@
+ * presented, each at a different address. This function is told which slot is active, and sets the
+ * update pointer to point to the correct non-active entry in the dual resource locator.
+ *
+ * @param[in, out] file		 pointer to dual resource locator with space separator. Note
```
Start the parameter definitions with capital letter.

NordicBuilder commented:
### Integration test specification

| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-ci-nrfconnect-boot-fw-update | X | | |
| test-fw-nrfconnect-chip | X | | |
| test-fw-nrfconnect-nrf-iot_mosh | X | | |
| test-fw-nrfconnect-zigbee | X | | |
| test-sdk-find-my | X | | |
| test-sdk-homekit | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf/job/master/25657/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

peknis commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -486,7 +486,9 @@ Bluetooth libraries and services
-|no_changes_yet_note|
+* :ref:`lib_dfu_target` library:
+
+   * Moved :c:func:`dfu_ctx_mcuboot_set_b1_file` to :ref:`lib_fota_download` and renamed to :c:func:`fota_download_parse_dual_resource_locator`
```
```suggestion
   * Moved the :c:func:`dfu_ctx_mcuboot_set_b1_file` function to :ref:`lib_fota_download` and renamed to :c:func:`fota_download_parse_dual_resource_locator`.
```

hakonfam reviewed (changes requested):
We are missing a `testcase.yaml` new tests. Without this twister won't pick up the tests.

Also, are there any function changes here, or is this just a refactor?

glarsennordic commented:
Library-wise, no function changes, the function was simply moved and renamed.

However, I somewhat significantly altered the functionality of the existing `fota_download` and `dfu_ctx_mcuboot_set_b1_file` tests. 

`dfu_ctx_mcuboot_set_b1_file` tests lost some supported platforms due to their integration into the `fota_download` test suite.

I modularized and increased the case coverage for both the `fota_download` and `dfu_ctx_mcuboot_set_b1_file` tests.

To facilitate testing, I separated the function into a `_utils.c` and `_utils.h` file. 

I also updated the test suite to use the new Zephyr 2 APIs.

As for the missing testcase.yaml, I meant to delete the entire `dfu_target_mcuboot` folder, since the only tests that had been in there were the `dfu_ctx_mcuboot_set_b1_file` which got moved to a new suite. Looks like I had a rebase go a little strangely on me and resurrect the folder. Sorry about that. I will get that deleted real fast.

glarsennordic commented:
@joerchan Adding you as a reviewer, since your commit 2634f67925dcde809c393c02590dfed0b636da28 / PR https://github.com/nrfconnect/sdk-nrf/pull/8375 was merged before this could be merged

I want to make sure that my rebase did not break your tweak.

glarsennordic commented on tests/subsys/net/lib/fota_download/src/test_fota_download.c:
```diff
@@ -11,26 +12,20 @@
 #include <fw_info.h>
 #include <pm_config.h>
 #include <fota_download.h>
+#include "test_fota_download_common.h"
```
Weird, I have this line in my copy of the source. Not sure why it isn't in what got pushed to github?

glarsennordic replied:
Good catch

tony-le-24 commented on tests/subsys/net/lib/fota_download/src/test_parse_dual_resource_locator.c:
```diff
@@ -0,0 +1,172 @@
+	 * no mechanism in place do decude if an address maps to non-volatile
+	 * storage or RAM
+	 */
+	ztest_test_skip();
```
Friendly suggestion: ZTEST new API has a shorter way of skipping tests, you can use: `Z_TEST_SKIP_IFNDEF(CONFIG_[something]);` or ``Z_TEST_SKIP_IFDEF(CONFIG_[something]);` and do not have to make another copy of the same test function just to skip it.

glarsennordic replied:
Neat, I will adopt this.

In fact, we have flat out dropped `NATIVE_POSIX` support from these tests, so the entire check could be removed. But I think it is harmless to allow such a simple skip line to remain in the code-base in case `NATIVE_POSIX` is ever re-added.

sigvartmh commented:
From CI it looks like this `Compilation of project 'secure_services_nrf9160dk_nrf9160_ns_gcc' failed with error code 'CompileErrors.CompilationError'` is the issue. Tried to compile it locally, but I didn't get an error @joan-nordic  could you provide the parameters it's using for building or follow up on this error?

joan-nordic commented:
@sigvartmh this sample has been removed in main, and we were building it on boot-update domain, this has been fixed and a succesfull run showed up on Jenkins after rebuilding, but GH did not update its status since it was no triggered from the PR itself. Please rerun. Thanks.

sigvartmh commented:
@glarsennordic ^

glarsennordic commented:
@sigvartmh @joan-nordic Thanks for looking into my CI troubles!

And thank you to whoever triggered a re-run, and to @gmarull  for getting this merged while I was out of the office! Really appreciate it.
